### ISSUE TITLE
file_exists() and touch() does not work

### ISSUE THREAD

**Issue by chippyash** [mikey179/vfsStream#98]
Checked known and resolved issues - not in there.

info:
 php -v
PHP 5.5.9-1ubuntu4.5 (cli) (built: Oct 29 2014 11:59:10) 

<pre>
if (!file_exists($fileName)) {
     touch($fileName);
}
</pre>

where $fileName == "vfs://digest.txt"

gives error 
PHP Fatal error:  Call to a member function lastModified() on a non-object in ..../vendor/mikey179/vfsStream/src/main/php/org/bovigo/vfs/vfsStreamWrapper.php on line 488

This relates to the touch($fileName); line
but, clearly the file_exists() method is not working either.

code to create the file is:
<pre>
        $root = vfsStream::setup();
        vfsStream::newFile('digest.txt')->at($root)->setContent('');
        return vfsStream::url('digest.txt');
</pre>

Now, it's very possible that I've missed something as the API has changed considerably recently - so happy to be guided.

**Comment by sndsgd** [comment]
The root directory is named. The equivalent to the unix root directory ```/``` in vfsStream is ```vfs://root/```, so in your case, the path you'd want to use would be ```vfs://root/digest.txt```.

```php
$root = vfsStream::setup();
var_dump($root->url());
// => string(10) "vfs://root"
var_dump(vfsStream::url('root/digest.txt'));
// => string(21) "vfs://root/digest.txt"
```
You can also give the root directory a custom name if you so desire:
```php
$root = vfsStream::setup('your-custom-name');
var_dump($root->url());
// => string(22) "vfs://your-custom-name"
```
This PHPUnit test case illustrates how to create a path and pass it to built-in functions:
```php
<?php
use \org\bovigo\vfs\vfsStream;

class Issue98Test extends PHPUnit_Framework_TestCase
{
    protected $path;

    public function setUp()
    {
        vfsStream::setup();
        $this->path = vfsStream::url('root/issue98.txt');
    }

    public function testTouch()
    {
        $this->assertFalse(file_exists($this->path));
        $this->assertTrue(touch($this->path));
        $this->assertTrue(file_exists($this->path));
    }

    public function testContents()
    {
        $content = 'test content';
        $this->assertFalse(file_exists($this->path));
        $this->assertEquals(strlen($content), file_put_contents($this->path, $content));
        $this->assertTrue(file_exists($this->path));
        $this->assertEquals($content, file_get_contents($this->path));
    }
}
```

**Comment by chippyash** [comment]
Thanks for that, but I guess I am more concerned about the documentation.  I took my first cut of

<pre>
        $root = vfsStream::setup();
        vfsStream::newFile('digest.txt')->at($root)->setContent('');
        return vfsStream::url('digest.txt');
</pre>

from the documentation.  This is what caused the original error.  I replaced it with 

<pre>
        $root = vfsStream::setup();
        $file = vfsStream::newFile('digest.txt')->at($root)->setContent('');
        return $file->url();
</pre>

and that fixed my problem, leaving only a known issue as I was was using LOCK_EX in a file_put_contents call.

However, your point about the default root being vfs:://root/ is interesting and worth remembering.  Thanks.
